src/cider/nrepl/middleware/out.clj

```diff
@@ -50,17 +52,42 @@ Please do not inline; they must not be recomputed at runtime."}
   [messages type]
   (PrintWriter. (proxy [Writer] []
                   (close [] (.flush ^Writer this))
+                  ;; unfortunately we can't type hint the method argument
+                  ;; as `int` and `char[]` aren't supported by proxy.
                   (write
                     ([x]
-                     (.write (original-output type) x)
                      (with-out-binding [printer messages type]
-                       (.write printer x)))
+                       (cond
+                         (string? x)
+                         (do
+                           (.write ^Writer (original-output type) ^String x)
```

Comment:
Extra request (sorry for the pickiness), please verify that there's coverage for all these branches or add it

Context: touching this area recently resulted in an opaque issue that impacted users.